### ISSUE TITLE
Source Postgres: Incremental sync failing with raw state retrieval not supported error after Airbyte upgrade

### ISSUE THREAD

**Issue by muntaserkhan-branch** [airbytehq/airbyte#30998]
### Connector Name

Postgres

### Connector Version

3.1.7

### What step the error happened?

During the sync

### Revelant information

We are in the process of upgrading Airbyte to 0.50.28 to use Xmin benefits. When I was testing postgres sync to our S3 Datalake, I am getting the following raw state retrieval not supported error posted in relevant log output. Postgres connector version is 3.1.7. I tested the older Postgres 1.0.11 connector also (which is in use in our current airbyte version 0.40.10), but I get the same error.  I have also tested both Detect Changes with Xmin System Column and Scan Changes with User Defined Cursor configurations during connection setup to see if it only happens for Xmin, but it happens for non Xmin also

### Relevant log output

```shell
2023-10-02 21:11:48 source > INFO i.a.i.s.p.PostgresUtils(isXmin):165 using Xmin: true
2023-10-02 21:11:48 source > ERROR i.a.i.b.s.SshWrappedSource(read):77 Exception occurred while getting the delegate read iterator, closing SSH tunnel java.lang.UnsupportedOperationException: Raw state retrieval not supported by global state manager.
        at io.airbyte.integrations.source.relationaldb.state.LegacyStateManager.getRawStateMessages(LegacyStateManager.java:95) ~[io.airbyte.airbyte-integrations.connectors-source-relational-db-24.0.2.jar:?]
        at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:530) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-24.0.2.jar:?]
        at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:121) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-24.0.2.jar:?]
        at io.airbyte.integrations.source.relationaldb.AbstractDbSource.read(AbstractDbSource.java:161) ~[io.airbyte.airbyte-integrations.connectors-source-relational-db-24.0.2.jar:?]
        at io.airbyte.integrations.base.ssh.SshWrappedSource.read(SshWrappedSource.java:75) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.base.IntegrationRunner.readSerial(IntegrationRunner.java:269) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:168) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:123) ~[io.airbyte.airbyte-integrations.bases-base-java-24.0.2.jar:?]
        at io.airbyte.integrations.source.postgres.PostgresSource.main(PostgresSource.java:683) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-24.0.2.jar:?]
Stack Trace: java.lang.UnsupportedOperationException: Raw state retrieval not supported by global state manager.
        at io.airbyte.integrations.source.relationaldb.state.LegacyStateManager.getRawStateMessages(LegacyStateManager.java:95)
        at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:530)
        at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:121)
        at io.airbyte.integrations.source.relationaldb.AbstractDbSource.read(AbstractDbSource.java:161)
        at io.airbyte.integrations.base.ssh.SshWrappedSource.read(SshWrappedSource.java:75)
        at io.airbyte.integrations.base.IntegrationRunner.readSerial(IntegrationRunner.java:269)
        at io.airbyte.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:168)
        at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:123)
        at io.airbyte.integrations.source.postgres.PostgresSource.main(PostgresSource.java:683)
```

### Contribute

- [X] Yes, I want to contribute

**Comment by glevineLeap** [comment]
Anyone know what the most recent stable version is with regards to this error?

**Comment by glevineLeap** [comment]
FYI, I had a similar issue and fixed it by downgrading Airbyte to 0.50.31 per https://airbytehq.slack.com/archives/C021JANJ6TY/p1698172688700949?thread_ts=1698165780.336079&cid=C021JANJ6TY

**Comment by muntaserkhan-branch** [comment]
@glevineLeap I upgraded(as I was in 0.50.28) to 0.50.31 and still get the same error while ingesting from postgres to data lake.
` java.lang.UnsupportedOperationException: Raw state retrieval not supported by global state manager.
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.source.relationaldb.state.LegacyStateManager.getRawStateMessages(LegacyStateManager.java:95) ~[airbyte-cdk-db-sources-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.integrations.source.postgres.xmin.XminCtidUtils.categoriseStreams(XminCtidUtils.java:46) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-0.50.30.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:466) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-0.50.30.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.integrations.source.postgres.PostgresSource.getIncrementalIterators(PostgresSource.java:121) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-0.50.30.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.source.relationaldb.AbstractDbSource.read(AbstractDbSource.java:161) ~[airbyte-cdk-db-sources-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.base.ssh.SshWrappedSource.read(SshWrappedSource.java:75) ~[airbyte-cdk-core-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.base.IntegrationRunner.readSerial(IntegrationRunner.java:271) ~[airbyte-cdk-core-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.base.IntegrationRunner.runInternal(IntegrationRunner.java:170) ~[airbyte-cdk-core-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.cdk.integrations.base.IntegrationRunner.run(IntegrationRunner.java:125) ~[airbyte-cdk-core-0.1.9.jar:?]
2023-10-26 03:36:33 source >         at io.airbyte.integrations.source.postgres.PostgresSource.main(PostgresSource.java:692) ~[io.airbyte.airbyte-integrations.connectors-source-postgres-0.50.30.jar:?]`

**Comment by muntaserkhan-branch** [comment]
resolved in later version 0.50.56
